auacm/app/static/js/sockets.js

```diff
@@ -0,0 +1,14 @@
+function Socket(url) {
+    this.ws = new WebSocket(url);
+    this.events = {};
+    var parent = this;
+    this.on = function(event, func) {
+        parent.events[event] = func;
+    };
+    this.ws.onmessage = function(event) {
+        var data = JSON.parse(event.data);
+        if (data.eventType in parent.events) {
+            parent.events[data.eventType](data.data);
+        }
+    };
+}
```

Comment:
Is this likely to be expanded in the future? If not, I don't see the real need to put it in a separate file and create the overhead of _another_ request to the server.